include/drjit-core/jit.h

```diff
@@ -494,9 +494,14 @@ extern JIT_EXPORT void jit_registry_remove(const void *ptr);
 extern JIT_EXPORT uint32_t jit_registry_id(const void *ptr);
 
 /// Return the largest instance ID for the given domain
+/// If the \c domain is a nullptr, it returns the largest instance ID for the given backend
 extern JIT_EXPORT uint32_t jit_registry_id_bound(JitBackend backend,
                                                  const char *domain);
 
+/// Fills the \c dest pointer array with all pointers registered in the registry
+/// \c dest has to point to an array with \c jit_registry_id_bound(backend, nullptr) entries
+extern JIT_EXPORT void jit_registry_fill_ptrs(JitBackend backend, void **dest);
```

Comment:
"fill_ptrs" could sound like you're filling the registry. Maybe `jit_registry_all_pointers()` or `jit_registry_get_pointers()`?